modules/Classroom/components/MaterialFormModal/MaterialFormModal.tsx

```diff
@@ -16,48 +16,88 @@ import { showNotification } from "@mantine/notifications";
 import { useForm, Controller } from "react-hook-form";
 
 import { ApiError } from "@/modules/Login/containers/LoginContainer.types";
-import { useUploadResourceMutation } from "@/shared/redux/rtk-apis/classworks/classworks.api";
+import {
+  useUploadResourceMutation,
+  useUpdateResourceMutation,
+} from "@/shared/redux/rtk-apis/classworks/classworks.api";
 
-import { materialFormSchema, TMaterialFormValues } from "./MaterialFormModal.helpers";
+import {
+  materialFormSchema,
+  TMaterialFormValues,
+  materialFormSchemaEdit,
+} from "./MaterialFormModal.helpers";
 import { IMaterialFormModalProps } from "./MaterialFormModal.interface";
 import { inputStyles } from "./MaterialFormModal.styles";
 
-const MaterialFormModal: React.FC<IMaterialFormModalProps> = ({ opened, onClose, classroomId }) => {
+const MaterialFormModal: React.FC<IMaterialFormModalProps> = ({
+  opened,
+  onClose,
+  classroomId,
+  resource,
+}) => {
   const {
     control,
     handleSubmit,
     register,
     formState: { errors },
     reset,
+    setValue,
   } = useForm<TMaterialFormValues>({
-    resolver: zodResolver(materialFormSchema),
+    resolver: zodResolver(resource ? materialFormSchemaEdit : materialFormSchema),
     mode: "onBlur",
   });
 
   const [isLoading, setIsLoading] = useState(false);
   const [uploadResource] = useUploadResourceMutation();
+  const [updateResource] = useUpdateResourceMutation();
+
+  useEffect(() => {
+    if (resource) {
+      setValue("title", resource.title);
+      setValue("description", resource.description);
```

Comment:
Are you using this to set default values to the form?

Reply:
Yes, when user clicks on the edit it will set the default values.